client/state/data-layer/wpcom/sites/comments/change-status/index.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { COMMENTS_CHANGE_STATUS } from 'state/action-types';
+import { mergeHandlers } from 'state/action-watchers/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { local } from 'state/data-layer/utils';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { errorNotice, removeNotice } from 'state/notices/actions';
+import { getSiteComment } from 'state/selectors';
+
+const changeCommentStatus = ( { dispatch, getState }, action ) => {
+	const { siteId, commentId, status } = action;
+	const previousStatus = get( getSiteComment( getState(), action.siteId, action.commentId ), 'status' );
+
+	dispatch(
+		http(
+			{
+				method: 'POST',
+				path: `/sites/${ siteId }/comments/${ commentId }`,
+				apiVersion: '1.1',
+				body: {
+					status,
+				},
+			},
+			{
+				...action,
+				previousStatus,
+			}
```

Comment:
I think the purpose here is to augment the original action with supplemental information